pkg/controller/certificates/sync.go

```diff
@@ -856,3 +873,16 @@ func setSecretValues(ctx context.Context, crt *cmapi.Certificate, s *corev1.Secr
 
 	return nil
 }
+
+func bytesEqual(l, r []byte) bool {
```

Comment:
Go provides `bytes.Equal`, any reason we use this function instead?

Reply:
>That functions actually uses bytes.Compare as I had not spotted 'Equal'! I
chose to have a specific function for it in order to explicitly handle the
'nil' cases...

I see that becomes an issue in bytes.Equal
https://play.golang.org/p/o9JgwH2DaCO